src/files/XmlFile.spec.ts

```diff
@@ -86,69 +87,114 @@ describe('XmlFile', () => {
             expect(file.scriptTagImports).to.be.lengthOf(3);
             expect(file.scriptTagImports[0]).to.deep.include(<FileReference>{
                 text: 'ChildScene1.brs',
-                filePathRange: Range.create(3, 58, 3, 73)
+                filePathRange: Range.create(2, 42, 2, 57)
             });
             expect(file.scriptTagImports[1]).to.deep.include(<FileReference>{
                 text: 'ChildScene2.brs',
-                filePathRange: Range.create(3, 116, 3, 131)
+                filePathRange: Range.create(2, 100, 2, 115)
             });
             expect(file.scriptTagImports[2]).to.deep.include(<FileReference>{
                 text: 'ChildScene3.brs',
-                filePathRange: Range.create(3, 174, 3, 189)
+                filePathRange: Range.create(2, 158, 2, 173)
             });
         });
 
-        it('finds component names', async () => {
+        it('finds component names', () => {
             file = new XmlFile('abs', 'rel', program);
-            await file.parse(`
+            file.parse(trim`
                 <?xml version="1.0" encoding="utf-8" ?>
                 <component name="ChildScene" extends="ParentScene">
-                <script type="text/brightscript" uri="ChildScene.brs" />
+                    <script type="text/brightscript" uri="ChildScene.brs" />
                 </component>
             `);
             expect(file.parentComponentName).to.equal('ParentScene');
             expect(file.componentName).to.equal('ChildScene');
         });
 
-        it('Adds error when no component is declared in xml', async () => {
+        it('Adds error when whitespace appears before the prolog', () => {
             file = new XmlFile('abs', 'rel', program);
-            await file.parse('<script type="text/brightscript" uri="ChildScene.brs" />');
+            file.parse(/* not trimmed */`
+                <?xml version="1.0" encoding="utf-8" ?>
+                <component name="ChildScene" extends="ParentScene">
+                    <script type="text/brightscript" uri="ChildScene.brs" />
+                </component>`
+            );
+            expect(file.diagnostics).to.be.lengthOf(2);
+            expect(file.diagnostics[0]).to.deep.include({ // expecting opening tag but got prolog
+                code: 1008,
+                range: Range.create(1, 16, 1, 22)
+            });
+            expect(file.diagnostics[1]).to.deep.include({
+                code: 1008,
```

Comment:
Rather than hardcoding the `1008` code here, use the spread operator on `DiagnosticMessages.xmlGenericParseError`. 

```
expect(file.diagnostics[1]).to.deep.include({
   ...DiagnosticMessages.xmlGenericParseError('Syntax error: whitespace found before the XML prolog'),
   range: Range.create(0, 0, 1, 16)
});
```